x/assets/keeper/params.go

```diff
@@ -39,22 +39,26 @@ func (k Keeper) GetParams(ctx sdk.Context) (*assetstypes.Params, error) {
 	return ret, nil
 }
 
-func (k Keeper) GetExocoreGatewayAddress(ctx sdk.Context) (common.Address, error) {
+func (k Keeper) GetGatewayAddresses(ctx sdk.Context) ([]common.Address, error) {
 	param, err := k.GetParams(ctx)
 	if err != nil {
-		return common.Address{}, err
+		return []common.Address{}, err
+	}
+	gateways := []common.Address{}
+	for _, gateway := range param.Gateways {
+		gateways = append(gateways, common.HexToAddress(gateway))
 	}
-	return common.HexToAddress(param.ExocoreLzAppAddress), nil
+	return gateways, nil
 }
 
-func (k Keeper) CheckExocoreGatewayAddr(ctx sdk.Context, addr common.Address) error {
+func (k Keeper) IsAuthorizedGateway(ctx sdk.Context, addr common.Address) error {
```

Comment:
Should there be a mapping from client chain id to gateway? Then, gateways could be validated on a per-chain basis.

Reply:
sounds reasonable, but it might have some potential drawbacks, i.e. : make it more complicated to set authorized gateway, since that should be done per chain, and introduce non-determinism if we use map in proto?